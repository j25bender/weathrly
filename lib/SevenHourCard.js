// import React, { Component } from 'react';
// import '../styles/SevenHourCard.css';

// export default class SevenHourCard extends React.Component {
//     constructor() {
//         super();        
//     }

//     render () {
//         return (
//             <div className="sevenHourCard">
//                 <div>{ this.props.data.hour }</div>
//                 <div>{ this.props.data.temp }ËšF</div>
//                 <img className="weather-icon" src={ this.props.data.icon_url } />
//             </div>
//         )
//     }
// }
