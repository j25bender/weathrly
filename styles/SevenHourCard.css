.cards {
    border: 0.01rem solid ghostwhite;    
    margin: 1rem;
    min-width: 5.5rem;
    padding: 1rem;
}

.card {
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;        
}

.sevenHourCard {
    border: 0.01rem solid ghostwhite;
    margin: 1rem;
    max-width: 15rem;
    min-width: 5.5rem;
    padding: 1rem;
  }
  
  .temp {
    margin: 0;
  }

  .sevenHourHeader {
    color: #fff;
    font-family: "Open Sans";
    font-size: 1.8rem;
    padding: 1rem 0 1rem 1rem;
    text-align: left;
  }
  
  @media screen and (max-width: 1010px) {
    .card {
      align-content: flex-start;
      flex-wrap: wrap;
      min-width: 15rem;
    }
  }