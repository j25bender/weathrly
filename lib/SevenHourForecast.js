import React from 'react';
import SevenHourCard from './SevenHourCard.js';

function SevenHourForecast (props) {
  return (
    <div className="cards">
      <div className="sevenHourHeader">
        <h3>7 Hour Forecast</h3>
      </div>
      <div className="card">
      {
         props.forecast.map( (card, index) => <SevenHourCard key={index} data={card} /> )
      }
      </div>
    </div>
  )
}

export default SevenHourForecast;