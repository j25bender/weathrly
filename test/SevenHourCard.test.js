import React from 'react';
import { shallow, mount } from 'enzyme';
import SevenHourForecast from '../lib/SevenHourForecast';
import SevenHourCard from '../lib/SevenHourCard';
import cleanData from '../lib/cleanData';
import mockData from '../lib/mockData';
import 'jest-localstorage-mock';

describe('SevenHourForecast Tests ', () => {
    let filteredData;
    let wrapper;
    
    beforeEach(() => {
        filteredData = cleanData(mockData);
        wrapper = mount(<SevenHourForecast forecast={filteredData.sevenHourForecastArray}/>);
    });

    it('SevenHourForecast component should exist', () => {
        expect(wrapper).toBeDefined()
        expect(wrapper.find('h3').text()).toEqual('7 Hour Forecast')        
    });

    it('SevenHourForecast should be a function', () => {
        expect(typeof SevenHourForecast).toEqual('function');
    });

    it('SevenHourForecast should render 7 SevenHourCard components', () => {
        expect(wrapper.find('SevenHourCard').length).toEqual(7);
        expect(SevenHourCard).toMatchSnapshot();
    });
  
    it('Each SevenHourCard component should render an hour, temp and an icon via an image url', () => {
        const firstCard = wrapper.find('SevenHourCard').first()
        const lastCard = wrapper.find('SevenHourCard').last()

        expect(firstCard.props().data.hour).toEqual('12:00 PM')
        expect(lastCard.props().data.hour).toEqual('6:00 PM')

        expect(firstCard.props().data.temp).toEqual('47')
        expect(lastCard.props().data.temp).toEqual('45')

        expect(wrapper.find('.weather-icon').length).toEqual(7);
        expect(firstCard.props().data.icon_url).toEqual('http://icons.wxug.com/i/c/k/partlycloudy.gif')
        expect(lastCard.props().data.icon_url).toEqual('http://icons.wxug.com/i/c/k/nt_clear.gif')
    });
});