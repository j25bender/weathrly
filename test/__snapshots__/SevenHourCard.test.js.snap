// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SevenHourForecast Tests  SevenHourForecast should render 7 SevenHourCard components 1`] = `[Function]`;
